### ISSUE TITLE
Reproduce the results

### ISSUE THREAD

**Issue by nixingyang** [GuoShi28/2StageAlign#2]
Thanks for publishing the source code.
I want to train a model from scratch and reproduce the results reported in the paper.
- Does the [batch_size](https://github.com/GuoShi28/2StageAlign/blob/7abfdbfd6248fbbb1aeae359cf658c5a445c9f40/codes/options/train/train_burst_JDD_2stage.yml#L24) refer to the total amount of samples in one iteration? In other words, is it the batch size in one GPU multiplied by the number of GPUs?
- I am using one GPU at the moment. It appears that the `batch_size` must be set to 1 in this case. Does the existing code support using a larger batch size?

**Comment by nixingyang** [comment]
For the record, the existing implementation does not support using a batch size rather than 1.
This is due to how the BiGRU module works in the training procedure, and it can be easily fixed by duplicating the entries in `gains` in `apply_gains_jdd` and `ccms` in `apply_ccms`.
There is no need to use two RTX 2080 Ti GPUs, and one should set a higher value for batch size in order to saturate the GPUs.

**Comment by GuoShi28** [comment]
Sorry, I just read your message.
1. yes, the batch size refers to the total amount of samples in one iteration.
2. For the batch larger than 1. Thank you for your effort, I will refer to your suggestion and correct my code. Thank you.
If there are other problems in reproducing the results, please feel free to contact me.

**Comment by ZcsrenlongZ** [comment]
> For the record, the existing implementation does not support using a batch size rather than 1. This is due to how the BiGRU module works in the training procedure, and it can be easily fixed by duplicating the entries in `gains` in `apply_gains_jdd` and `ccms` in `apply_ccms`. There is no need to use two RTX 2080 Ti GPUs, and one should set a higher value for batch size in order to saturate the GPUs.

Hi! How long time do you take when training from scratch?

**Comment by GuoShi28** [comment]
> > For the record, the existing implementation does not support using a batch size rather than 1. This is due to how the BiGRU module works in the training procedure, and it can be easily fixed by duplicating the entries in `gains` in `apply_gains_jdd` and `ccms` in `apply_ccms`. There is no need to use two RTX 2080 Ti GPUs, and one should set a higher value for batch size in order to saturate the GPUs.
> 
> Hi! How long time do you take when training from scratch?

I remeber use about 3 days (60W iteration) to train a model in my oridinary experiment setting.

**Comment by ZcsrenlongZ** [comment]
> > > For the record, the existing implementation does not support using a batch size rather than 1. This is due to how the BiGRU module works in the training procedure, and it can be easily fixed by duplicating the entries in `gains` in `apply_gains_jdd` and `ccms` in `apply_ccms`. There is no need to use two RTX 2080 Ti GPUs, and one should set a higher value for batch size in order to saturate the GPUs.
> > 
> > 
> > Hi! How long time do you take when training from scratch?
> 
> I remeber use about 3 days (60W iteration) to train a model in my oridinary experiment setting.

Thanks for your reply! but when I training from scratch in 2080Ti, I found it is slow , and I try to set num_works larger,but it doesn't works, do you have any advice?

**Comment by nixingyang** [comment]
The existing implementation is not efficient. One may inspect the GPU utilization during the training.
You could increase the batch size and learning rate. Optimization of the data loader would also help.
As a side note, I am able to reproduce the authors' results on REDS4.

**Comment by GuoShi28** [comment]
> The existing implementation is not efficient. One may inspect the GPU utilization during the training. You could increase the batch size and learning rate. Optimization of the data loader would also help. As a side note, I am able to reproduce the authors' results on REDS4.

Thank you for your suggestion.

**Comment by GuoShi28** [comment]
> > > > For the record, the existing implementation does not support using a batch size rather than 1. This is due to how the BiGRU module works in the training procedure, and it can be easily fixed by duplicating the entries in `gains` in `apply_gains_jdd` and `ccms` in `apply_ccms`. There is no need to use two RTX 2080 Ti GPUs, and one should set a higher value for batch size in order to saturate the GPUs.
> > > 
> > > 
> > > Hi! How long time do you take when training from scratch?
> > 
> > 
> > I remeber use about 3 days (60W iteration) to train a model in my oridinary experiment setting.
> 
> Thanks for your reply! but when I training from scratch in 2080Ti, I found it is slow , and I try to set num_works larger,but it doesn't works, do you have any advice?

I also use 2080Ti in my experiments. You can refer to (https://github.com/GuoShi28/2StageAlign/issues/2#issuecomment-1273674720) for increasing batch size and learning rate. I will check how to speed up the training process in my future work. Thank you.
